pkg/scheduler/resource/relationship_check_task.go

```diff
@@ -299,5 +341,56 @@ func (in *RelationshipCheckTask) setResourceStatusFalse(
 		return nil
 	}
 
+	if pkgresource.IsStatusStopped(parentResource) {
+		status.ResourceStatusProgressing.False(
+			res,
+			fmt.Sprintf("Parent resource status is stopped, name: %s", parentResource.Name),
+		)
+
+		err := pkgresource.UpdateStatus(ctx, in.modelClient, res)
+		if err != nil {
+			return err
+		}
+
+		return nil
+	}
+
 	return nil
 }
+
+// checkDependantResourceStatus sets a resource status to false if dependant resource status is false or deployed.
+func (in *RelationshipCheckTask) checkDependantResourceStatus(
+	ctx context.Context,
+	res, dependantResource *model.Resource,
+) (bool, error) {
+	if pkgresource.IsStatusFalse(dependantResource) {
```

Comment:
IsStatusFalse is confusing, could you update to IsStatusFailed or IsStatusError

Reply:
updated